### DISCUSSION TITLE
Release 0.9 migration guide

### DISCUSSION THREAD

**Discussion by EdgyEdgemond** [EdgyEdgemond/changelog-gen#98] [Announcements]
With 0.9.x a few things have changed. The majority of features have been maintained for backwards compatibility.

This release brings with it some shiny new features:

* support for [Conventional Commit](https://www.conventionalcommits.org/en/v1.0.0/) parsing.
* support for configuration in `pyproject.toml`
* verbose debugging `-v`, `-vvv` etc
* expanded `post_process` support, including github issues

# Breaking Changes

* Python 3.8 support has been dropped.
* Use of `post_process` requires `headers` to be defined in configuration.

# Migrating from 0.8

First step is to update the dependency pin, to maintain existing functionality using setup.cfg and bump2version (development is dead on this project). Update the pin to `0.9.0` and use the extras flag `bump2version`

```
changelog-gen = { version = "~0.9.0", extras=["bump2version"]}
```

All existing configuration should continue to work.

## Migrating to pyproject.toml configuration

Run the command `changelog migrate` and copy the provided `toml` configuration into your pyproject.toml file, remove `[changelog]` section from setup.cfg.

When running `changelog generate` (replacing legacy command `changelog-gen`. There may be some deprecation warning depending on your configuration, these can be silenced by following the suggestions in the warning.

## Migrating to bump-my-version

`bump-my-version` is the new active fork of bumpversion and supports pyproject.toml configuration.

To switch to this update the extras tag in your dependency from `bump2version` to `bump-my-version` and update.

For most configurations of bumpversion, it should be a simple conversion from setup.cfg `[bumpversion]` to pyproject.toml `[tool.bumpversion]`, check out [bump-my-version](https://github.com/callowayproject/bump-my-version) for details on any configuration issues that might pop up.

At this point you should be able to finally drop setup.cfg completely.

# Custom sections

If you are making use of custom sections, take a look at the `type_headers` configuration in the README, this replaces `sections` and `section_mappings` configuration.

**Comment by EdgyEdgemond** [comment]
See the [README](https://github.com/EdgyEdgemond/changelog-gen/blob/master/README.md) and [CHANGELOG](https://github.com/EdgyEdgemond/changelog-gen/blob/master/CHANGELOG.md) for more details on new features.

**Comment by EdgyEdgemond** [comment]
@LachlanMarnham @myddrin latest release is here, should be fully backwards compatible with 0.8.x.

**Reply by LachlanMarnham** [reply]
Thanks @EdgyEdgemond. There's a breaking change for py3.11 on Windows which is missed by the test matrix in CI. I'll create an issue shortly and open a PR this afternoon

**Reply by LachlanMarnham** [reply]
Ok decided not to open a PR for `changelog-gen` since the issue has been resolved already in `rtoml` and is awaiting release. See: https://github.com/EdgyEdgemond/changelog-gen/issues/103
